src/image_cache/image_loader.rs

```diff
@@ -150,6 +157,21 @@ pub fn load_gif(path: &Path, req_id: u32) -> Result<impl Iterator<Item = Result<
 	load_animation(req_id, decoder)
 }
 
+/// Parse, render and gather an SVG into a ImageBuffer<Rgba>
+pub fn load_svg(path: &std::path::Path) -> Result<image::RgbaImage> {
+	let opt = usvg::Options::default();
+	let rtree = usvg::Tree::from_file(path, &opt)?;
+	let (width, height) = rtree.svg_node().size.to_screen_size().dimensions();
+	// Scale to fit 4096
+	let zoom = 4096 / width.max(height);
+	let (width, height) = (width * zoom, height * zoom);
+
+	// These unwrapped Options are fine as long as the dimensions are correct
+	let mut pixmap = tiny_skia::Pixmap::new(width, height).unwrap();
+	resvg::render(&rtree, usvg::FitTo::Zoom(zoom as f32), pixmap.as_mut()).unwrap();
+	Ok(image::RgbaImage::from_raw(width, height, pixmap.data().to_vec()).unwrap())
```

Comment:
Yes, this is exactly what I was thinking about. 😊

There is one more improvement that can be made here. `pixmap.data()` only gives you a reference to the pixel array within `pixmap`, then `to_vec` takes the reference and creates a new array, copying all the data, then you throw away the entire array contained within `pixmap`. As you can see there's a copy which we don't really need. We can avoid creating a copy by *moving* the `pixmap`'s internal data into the `from_raw` function. Unfortunately the `data` inside `Pixmap` is not `pub` so we can't reach it directly. But there's a function for `Pixmap` which was created exatcly for this purpose, as shown below. (Notice that this function takes `self` instead of `&self`, this means that `self` is *moved*)
```rust
    /// Consumes the internal data.
    ///
    /// Bytes are ordered as RGBA.
    pub fn take(self) -> Vec<u8> {
        self.data
    }
```

Reply:
Thanks for the detailed pointers, they really helped. It should be fine now.